src/main/java/mate/academy/service/UserService.java

```diff
@@ -0,0 +1,15 @@
+package mate.academy.service;
+
+import java.util.List;
+import java.util.Optional;
+import mate.academy.model.User;
+
+public interface UserService {
+    User add(User user);
```

Comment:
Ensure that the logic for salt and password hashing is implemented in the `add()` method of the `UserService` implementation. This is a core requirement.